ledger-core/virtual/execute/execute.go

```diff
@@ -388,15 +391,97 @@ func (s *SMExecute) stepDeduplicateUsingPendingsTable(ctx smachine.ExecutionCont
 	}
 
 	if isDuplicate {
-		// TODO: do we have result? if we have result then we should resend result
-		ctx.Log().Warn("duplicate found as pending request")
-		return ctx.Stop()
+		if isActive {
+			ctx.Log().Warn("duplicate found as pending request")
+			return ctx.Stop()
+		}
 	}
 
 	if ctx.AcquireForThisStep(pendingListFilled).IsNotPassed() {
 		return ctx.Sleep().ThenRepeat()
 	}
 
+	return ctx.Jump(s.stepDeduplicateThroughPreviousExecutor)
+}
+
+type deduplicationBargeIn struct {
+	lookAt   pulse.Number
+	callee   reference.Global
+	outgoing reference.Global
+}
+
+func (s *SMExecute) stepDeduplicateThroughPreviousExecutor(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	msg := payload.VFindCallRequest{
+		LookAt:   s.Payload.CallOutgoing.GetPulseNumber(),
+		Callee:   s.execution.Object,
+		Outgoing: s.execution.Outgoing,
+	}
+
+	bargeInCallback := ctx.NewBargeInWithParam(func(param interface{}) smachine.BargeInCallbackFunc {
+		res, ok := param.(*payload.VFindCallResponse)
+		if !ok || res == nil {
+			panic(throw.IllegalValue())
+		}
+
+		return func(ctx smachine.BargeInContext) smachine.StateUpdate {
+			s.VFindCallResponse = res
+			return ctx.WakeUp()
+		}
+	})
+
+	bargeInKey := deduplicationBargeIn{
+		lookAt:   msg.LookAt,
+		callee:   msg.Callee,
+		outgoing: msg.Outgoing,
+	}
+
+	if !ctx.PublishGlobalAliasAndBargeIn(bargeInKey, bargeInCallback) {
+		return ctx.Error(throw.E("failed to publish bargeInCallback"))
+	}
+
+	s.messageSender.PrepareAsync(ctx, func(goCtx context.Context, svc messagesender.Service) smachine.AsyncResultFunc {
+		err := svc.SendRole(goCtx, &msg, node.DynamicRoleVirtualExecutor, s.execution.Object, s.pulseSlot.PulseData().PrevPulseNumber())
+		return func(ctx smachine.AsyncResultContext) {
+			if err != nil {
+				ctx.Log().Error("failed to send message", err)
+			}
+		}
+	}).WithoutAutoWakeUp().Start()
+
+	return ctx.Sleep().ThenJump(s.stepProcessFindCallResponse)
```

Comment:
probably, you need to add stepWaitFindCallResponse, you could be woken up by different signal. And you need to for s.VFindCallResponse != nil.

Reply:
yeah, miss it